tests/README.md

```diff
@@ -0,0 +1,79 @@
+## Backend development
+
+The backend test suite can be run outside the Docker container.
+
+### Start the services
+
+We need all but the main `backend` service running.  Since multiple `backend` instances are allowed, we will simply start everything up
+
+    # ENSURE THE IMAGES ARE CLEAN
+    docker-compose down
+    # SETUP ALL IMAGES
+    docker-compose up --build
+
+### Install into virtual environment
+
+Treeherder requires specific library versions that will likely interfere with what you have installed. It is best to setup a virtual environment to contain the quirks it introduces.
+
+Be sure you are in the `treeherder` main directory
+
+    pip install virtualenv
+    virtualenv venv
+    source venv/bin/activate
+    python -m pip install -r requirements/common.txt
+    python -m pip install -r requirements/dev.txt
+
+...or Windows...
+
+    pip install virtualenv
+    virtualenv venv
+    venv\Scripts\activate
+    python -m pip install -r requirements\common.txt
+    python -m pip install -r requirements\dev.txt
+
+### Set environment variables
+
+Treeherder requires a number of environment variables that point to the services.  In our case, those services are in local docker containers.
+
+    ./tests/env.sh     # env.bat for Windows
+
+If you plan to use an IDE, here is the same, as a very long line
+
+```
+BROKER_URL=localhost//guest:guest@rabbitmq//;DATABASE_URL=mysql://root@localhost:3306/treeherder;REDIS_URL=redis://localhost:6379;SITE_URL=http://backend:8000/;TREEHERDER_DEBUG=True;TREEHERDER_DJANGO_SECRET_KEY=secret-key-of-at-least-50-characters-to-pass-check-deploy;NEW_RELIC_DEVELOPER_MODE=True
```

Comment:
I filed this so contributors can take care of simplifying this:
https://github.com/mozilla/treeherder/issues/5926